src/workerd/jsg/fast-api.h

```diff
@@ -17,10 +17,21 @@
 #include <v8-value.h>
 
 #include <kj/common.h>
+#include <kj/string.h>
 
 namespace workerd::jsg {
 
+class ByteString;
+class DOMString;
 class Lock;
+class USVString;
+
+// Update this list whenever a new string type is added.
+template <typename T>
+concept StringLike =
+    kj::isSameType<kj::String, T>() || kj::isSameType<kj::ArrayPtr<const char>, T>() ||
+    kj::isSameType<kj::Array<const char>, T>() || kj::isSameType<ByteString, T>() ||
+    kj::isSameType<USVString, T>() || kj::isSameType<DOMString, T>();
```

Comment:
There's already a `isStringType` in `jsg/web-idl.h` ... this is also not accounting for the `jsg::NonCoercible` wrapper .. should it?